packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,144 @@
+import { Construct } from 'constructs';
+import { CfnRealtimeLogConfig } from './cloudfront.generated';
+import * as iam from '../../aws-iam';
+import * as kinesis from '../../aws-kinesis';
+import { IResource, Names, PhysicalName, Resource } from '../../core';
+
+/**
+ * @internal
+ */
+function singletonKinesisRole(scope: Construct): iam.IRole {
+  const id = 'RealtimeLogKinesisRole';
+  const existing = scope.node.tryFindChild(id) as iam.IRole;
+  if (existing) { return existing; }
+
+  const role = new iam.Role(scope, id, {
+    roleName: PhysicalName.GENERATE_IF_NEEDED,
+    assumedBy: new iam.ServicePrincipal('cloudfront.amazonaws.com'),
+  });
+
+  return role;
+}
+
+/**
+ * Represents the endpoints available for targetting within a realtime log config resource
+ */
+export abstract class Endpoint {
+  /**
+   * Configure a Kinesis Stream Endpoint for Realtime Log Config
+   *
+   * @default - a role will be created and used across your endpoints
+   */
+  public static fromKinesisStream(scope: Construct, stream: kinesis.IStream, role?: iam.IRole): Endpoint {
+    const cloudfrontRole = role || singletonKinesisRole(scope);
+
+    if (!role) {
```

Comment:
We can setup the permissions even if a role is passed

Reply:
Sure - will remove the conditional check